lib/classifier-reborn/bayes.rb

```diff
@@ -73,7 +73,9 @@ def train(category, text)
 
       @backend.update_category_training_count(category, 1)
       @backend.update_total_trainings(1)
-      Hasher.word_hash(text, @language, @enable_stemmer).each do |word, count|
+      word_hash = Hasher.word_hash(text, @language, @enable_stemmer)
+      return if word_hash.length == 0
```

Comment:
I would make it the very first thing in the method, even before calling `CategoryNamer`. This way, if the training has to be skipped, it does not make any changes in the data-structure state in any circumstances.